framework/wazuh/stats.py

```diff
@@ -82,6 +82,115 @@ def weekly():
     return result
 
 
+@expose_resources(actions=["agent:read"], resources=["agent:id:{agent_list}"],
+                  post_proc_kwargs={'exclude_codes': [1701, 1703, 1707]})
+def get_daemons_stats_agents(daemons_list: list = None, agent_list: list = None):
+    """Get agents statistical information from the specified daemons.
+    If the daemons list is empty, the stats from all daemons will be retrieved.
+    If the `all` keyword is included in the agents list, the stats from all the agents will be retrieved.
+
+    Parameters
+    ----------
+    daemons_list : list
+        List of the daemons to get statistical information from.
+    agent_list : list
+        List of agents ID's.
+
+    Returns
+    -------
+    AffectedItemsWazuhResult
+        Dictionary with daemon's statistical information of the specified agents.
+    """
+    daemon_socket_mapping = {'wazuh-remoted': common.REMOTED_SOCKET,
+                             'wazuh-analysisd': common.ANALYSISD_SOCKET}
+    result = AffectedItemsWazuhResult(all_msg='Statistical information for each daemon was successfully read',
+                                      some_msg='Could not read statistical information for some daemons',
+                                      none_msg='Could not read statistical information for any daemon')
+
+    if agent_list:
+        if 'all' not in agent_list:
+            system_agents = get_agents_info()
+            rbac_filters = get_rbac_filters(system_resources=system_agents, permitted_resources=agent_list)
+
+            with WazuhDBQueryAgents(limit=None, select=["id", "status"], **rbac_filters) as db_query:
+                data = db_query.run()
+
+            agent_list = set(agent_list)
+
+            try:
+                agent_list.remove('000')
+                result.add_failed_item('000', exception.WazuhError(1703))
+            except KeyError:
+                pass
```

Comment:
```suggestion
            with contextlib.suppress(KeyError):
                agent_list.remove('000')
                result.add_failed_item('000', exception.WazuhError(1703))
```

Reply:
Adding this suggestion. Not committing it directly as importing `contextlib` is needed.